models/benchmarks/himeno_model.py

```diff
@@ -73,13 +73,18 @@ def build_benchmark(self, compilers_dict, complete_compile_flags,
     def run_benchmark(self, binary_name, extra_runflags):
         """Runs the benchmarks using the base + extra flags"""
         binary_name = 'bmt'
-        binary_path = os.path.join(self.build_root, binary_name)
-        run_cmd = [[]]
-        if extra_runflags is None or extra_runflags == '':
-            run_cmd.append([binary_path])
-        else:
-            run_cmd.append([binary_path, extra_runflags])
-        return run_cmd
+        binary_path = os.path.join(self.benchmark_rootpath, self.name, binary_name)
+        run_cmds = []
+        run_args = self.parse_run_args(extra_runflags)
+        if (run_args.iterations).isdigit():
+            for i in range(0, int(run_args.iterations)):
+                run_cmd=[]
+                metadata = 'i=' + str(i) + 's=' + run_args.size + 'c=' + run_args.complexity
```

Comment:
This repeated logic will not scale as more benchmarks get introduced. We need to think of something else. Coupled with the fact that size/complexity won't scale either, I say we just forget for now and add only the id of the run.

Reply:
I don't really see how it wouldn't scale, but as you wish I guess....